docs/chapters/04_features.md

```diff
@@ -317,6 +317,69 @@ And this would be the response:
 
 > [!NOTE] Remember to set the proper **access token** for secured commands, check ["Authorizing operations"](#authorizing-operations).
 
+### Adding "before" hooks to your commands
+As seen above, when you send commands, you can include input parameters to be received for the command handler. Although, there are some cases where you'd want to do some actions just before the `handle` function gets called in your command, like:
+- Validating parameters
+- Change command inputs on the fly with custom logic
+- Deny access to specific users
+
+For these cases (and more), you can use the commands before hooks. 
+
+These hooks can be added in your @Command decorator through the `beforeCommand` parameter, as an array of functions:
+```typescript
+@Command({
+  authorize: [User],
+  beforeCommand: [ChangeCartItem.beforeFn],
+})
+export class ChangeCartItem {
+  public constructor(readonly cartId: UUID, readonly productId: UUID, readonly quantity: number) {
+  }
+
+  public static beforeFn(input: CommandInput, currentUser?: UserEnvelope): CommandInput {
+    if (input.cartUserId !== currentUser.id) {
+      throw NonAuthorizedUserException() // We don't let this user to trigger the command
+    }
+    return input
+  }
+}
+```
+
+As you can see, we just check if the `cartUserId` is equal to the `currentUser.id`, which is the user id extracted from the auth token. This way, we can throw an exception and avoid this user to call this command.
+
+But this is not all, remember that the `beforeCommand` parameter is an array of functions. That is because these functions are chainable:
+1. Imagine that we have 2 functions: `beforeFn` and `beforeFnV2`
+2. When sending a command, the `beforeFn` above will be called, and we could change the `input` variable (for example)
+3. That changed `input` variable will then be passed to the next function `beforeFnV2`
+
+Let's see it through an example:
+```typescript
+@Command({
+  authorize: [User],
+  beforeCommand: [ChangeCartItem.beforeFn, ChangeCartItem.beforeFnV2],
```

Comment:
Wondering: Why chaning the API here? I mean, in ReadModels we use `before`, but here we need to use `beforeCommand`, which is also redundant. Can't we use just use `before`?

Reply:
I just wanted to reuse the same `FilterHooks` interface for both but now looking at your second comment I think It's much clearer to keep it as `before` renaming also the interfaces. Thanks!!